vortex-array/src/arrays/decimal/compute/is_constant.rs

```diff
@@ -0,0 +1,61 @@
+use vortex_error::VortexResult;
+use vortex_scalar::i256;
+
+use crate::arrays::decimal::serde::DecimalValueType;
+use crate::arrays::{DecimalArray, DecimalEncoding, NativeDecimalType};
+use crate::compute::{IsConstantFn, IsConstantOpts};
+
+impl IsConstantFn<&DecimalArray> for DecimalEncoding {
+    fn is_constant(
+        &self,
+        array: &DecimalArray,
+        _opts: &IsConstantOpts,
+    ) -> VortexResult<Option<bool>> {
+        match array.values_type {
+            DecimalValueType::I128 => Ok(Some(compute_is_constant(&array.buffer::<i128>()))),
+            DecimalValueType::I256 => Ok(Some(compute_is_constant(&array.buffer::<i256>()))),
+        }
+    }
+}
+
+fn compute_is_constant<T: NativeDecimalType>(values: &[T]) -> bool {
+    // We know that the top-level `is_constant` ensures that the array is all_valid or non-null.
+    let first_value = values[0];
+
+    for &value in &values[1..] {
+        if value != first_value {
+            return false;
+        }
+    }
+
+    true
+}
```

Comment:
There's a very funky looking function for this in PrimitiveArray, you might want to use that. If we have AVX-512 it would be faster but gain here is definitely very small

Reply:
Yea I kinda figured since we're already working with at least 128-bit and possibly 256 bit types, the gains are not nearly as substantial as e.g. u32